File: server/models/roulette.go
package models

import (
    "time"

    "github.com/google/uuid"
)

type RouletteTable struct {
    ID        uuid.UUID `json:"id" db:"id"`
    Name      string    `json:"name" db:"name"`
    MinBet    float64   `json:"min_bet" db:"min_bet"`
    MaxBet    float64   `json:"max_bet" db:"max_bet"`
    Status    string    `json:"status" db:"status"`
    CreatedAt time.Time `json:"created_at" db:"created_at"`
}

type RouletteRound struct {
    ID            uuid.UUID  `json:"id" db:"id"`
    TableID       uuid.UUID  `json:"table_id" db:"table_id"`
    RoundNumber   int        `json:"round_number" db:"round_number"`
    Result        *int       `json:"result" db:"result"`
    ResultColor   *string    `json:"result_color" db:"result_color"`
    SeedHash      *string    `json:"seed_hash" db:"seed_hash"`
    SeedRevealed  *string    `json:"seed_revealed" db:"seed_revealed"`
    BettingEndsAt *time.Time `json:"betting_ends_at" db:"betting_ends_at"`
    CreatedAt     time.Time  `json:"created_at" db:"created_at"`
    SettledAt     *time.Time `json:"settled_at" db:"settled_at"`
}

type RouletteBet struct {
    ID        uuid.UUID `json:"id" db:"id"`
    RoundID   uuid.UUID `json:"round_id" db:"round_id"`
    UserID    uuid.UUID `json:"user_id" db:"user_id"`
    BetType   string    `json:"bet_type" db:"bet_type"`
    BetValue  string    `json:"bet_value" db:"bet_value"`
    Amount    float64   `json:"amount" db:"amount"`
    Payout    float64   `json:"payout" db:"payout"`
    Status    string    `json:"status" db:"status"`
    CreatedAt time.Time `json:"created_at" db:"created_at"`
}

File: client/types/roulette.ts
import { z } from "zod";

export type RouletteTableStatus = "active" | "inactive" | "maintenance";

export type RouletteTable = {
  readonly id: string;
  readonly name: string;
  readonly min_bet: string;
  readonly max_bet: string;
  readonly status: RouletteTableStatus;
  readonly created_at: string;
};

export type RouletteRound = {
  readonly id: string;
  readonly table_id: string;
  readonly round_number: number;
  readonly result: number | null;
  readonly result_color: string | null;
  readonly seed_hash: string | null;
  readonly seed_revealed: string | null;
  readonly betting_ends_at: string | null;
  readonly created_at: string;
  readonly settled_at: string | null;
};

export type RouletteBetStatus = "pending" | "won" | "lost";

export type RouletteBet = {
  readonly id: string;
  readonly round_id: string;
  readonly user_id: string;
  readonly bet_type: RouletteBetType;
  readonly bet_value: string;
  readonly amount: string;
  readonly payout: string;
  readonly status: RouletteBetStatus;
  readonly created_at: string;
};

export type RouletteBetType =
  | "straight"
  | "split"
  | "street"
  | "corner"
  | "line"
  | "dozen"
  | "column"
  | "red"
  | "black"
  | "odd"
  | "even"
  | "high"
  | "low";

export const placeBetSchema = z.object({
  round_id: z.string().uuid(),
  bet_type: z.enum([
    "straight",
    "split",
    "street",
    "corner",
    "line",
    "dozen",
    "column",
    "red",
    "black",
    "odd",
    "even",
    "high",
    "low",
  ]),
  bet_value: z.string().min(1),
  amount: z.string().regex(/^\d+(\.\d{1,2})?$/, "Invalid amount"),
});

export type PlaceBetInput = z.infer<typeof placeBetSchema>;
